synapse/lib/stormtypes.py

```diff
@@ -8017,6 +8021,18 @@ async def _methWipeLayer(self):
         view = self.runt.snap.core.getView(viewiden)
         await view.wipeLayer(useriden=useriden)
 
+    async def _methSwapLayer(self):
+
+        iden = self.valu.get('iden')
+        self.runt.reqAdmin(gateiden=iden)
+
+        view = self.runt.snap.core.getView(iden)
+        if not view.parent:
```

Comment:
why does it need to have a parent? (I'm not sure what the use case would be for doing it on a non-forked view but it seems possible)

Reply:
I mostly wanted to limit the scope because of the layer deletion and assumption of view admin perms being sufficient. We could consider some additional perms ( or opening up the use case ) later, just trying to keep the perms logic simple for now :+1: